app/src/main/java/com/eis/phoneringer/MainActivity.java

```diff
@@ -1,14 +1,127 @@
 package com.eis.phoneringer;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.Manifest;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.os.Build;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
 
+import com.eis.phoneringer.structure.AppManager;
+import com.eis.phoneringer.structure.PasswordManager;
+import com.eis.phoneringer.structure.ReceivedMessageListener;
+import com.eis.phoneringer.structure.RingCommand;
+import com.eis.smslibrary.SMSHandler;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.eis.smslibrary.exceptions.InvalidSMSMessageException;
+import com.eis.smslibrary.exceptions.InvalidTelephoneNumberException;
+import com.eis.smslibrary.listeners.SMSSentListener;
+
+/**
+ * Test app: through this MainActivity we can test out our basic library for sending ring commands
+ *
+ * @author Alberto Ursino
+ */
 public class MainActivity extends AppCompatActivity {
 
+    private static Button ringButton = null;
+    private static EditText phoneNumber = null;
+    private static final String DEFAULT_PASSWORD = "password";
+    private PasswordManager passwordManager = null;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        final Context context = getApplicationContext();
+        ringButton = findViewById(R.id.ring_button);
+        phoneNumber = findViewById(R.id.phone_number);
+        passwordManager = new PasswordManager(context);
+
+        /**
```

Comment:
Double asterisks are used to indicate a part of the javadoc for this class, regular multiline comments should use only one asterisk.

Reply:
Can't I use specifications for other parts of the code other than methods or classes?

Reply:
Double asterisk are used to tell the javadoc tool to add that comment to the documentation, but your comment is just a comment and not the specification of a method or a description of a class, so it should use only one asterisk in the first line

Reply:
Oh you are right, I will change. Thanks.